src/expander.wisp

```diff
@@ -335,3 +345,35 @@
   [& body]
   `(.call lazy-seq nil false (fn [] ~@body)))
 (install-macro :lazy-seq expand-lazy-seq)
+
+
+(defn expand-when
+  "Evaluates test. If logical true, evaluates body in an implicit do."
+  [test & body]
+  `(if ~test (do ~@body)))
+(install-macro :when expand-when)
+
+(defn expand-when-not
+  "Evaluates test. If logical false, evaluates body in an implicit do."
+  [test & body]
+  `(when (not ~test) ~@body))
+(install-macro :when-not expand-when-not)
+
+
+(defn expand-if-let
+  "bindings => binding-form test
+  body => [then else]
+  If test is true, evaluates then with binding-form bound to the value of
+  test, if not, yields else."
+  [bindings then else*]
+  (let [name (first bindings), test (second bindings), sym (gensym)]
+    `(let [~sym ~test]
+       (if ~sym (let [~name ~sym] ~then) ~else*))))
```

Comment:
is this just to avoid `~name` being visible in the else branch?

Reply:
I've been thinking of implementing [destructuring](https://clojure.org/guides/destructuring) some way or another, and this is how the binding should be done to be compatible with such a feature. And yes, in Clojure, the binding only happens in the then branch.
…Come to think of it, this was probably one of the reasons I forgo passing `name` to `gensym` (as when it's not a typeable symbol itself, generated output won't be valid JS).